test/opt/test_passes.cpp

```diff
@@ -0,0 +1,195 @@
+// running passes and checking results.
+//
+// To write value-Parameterized tests:
+//   using ValueParamTest = PassTest<::testing::TestWithParam<std::string>>;
+// To use as normal fixture:
+//   using FixtureTest = PassTest<::testing::Test>;
+template <typename GT>
+class PassTest : public GT {
+ public:
+  PassTest() : tools_(SPV_ENV_UNIVERSAL_1_1) {}
+
+  // Adds a pass to be run.
+  template <typename PassT>
+  void AddPass() {
+    manager_.AddPass<PassT>();
+  }
+
+  // Runs the pass on the binary assembled from the |original| assembly, and
+  // checks whether the optimized binary can be disassembled to the |expected|
+  // assembly.
+  void RunAndCheck(const std::string& original, const std::string& expected) {
+    assert(manager_.NumPasses());
+
+    std::unique_ptr<ir::Module> module = tools_.BuildModule(original);
+    ASSERT_NE(nullptr, module);
+
+    manager_.Run(module.get());
+
+    std::vector<uint32_t> binary;
+    module->ToBinary(&binary, /* skip_nop = */ false);
+
+    std::string optimized;
+    EXPECT_EQ(SPV_SUCCESS, tools_.Disassemble(binary, &optimized));
+    EXPECT_EQ(expected, optimized);
+  }
+
+ private:
+  SpvTools tools_;  // An instance for calling SPIRV-Tools functionalities.
+  opt::PassManager manager_;  // The pass manager.
+};
+
+using StripLineDebugInfoTest = PassTest<::testing::Test>;
+
+TEST_F(StripLineDebugInfoTest, LineNoLine) {
+  AddPass<opt::StripDebugInfoPass>();
+
+  std::vector<std::string> text = {
+      "OpCapability Shader",
+      "%1 = OpExtInstImport \"GLSL.std.450\"",
+      "OpMemoryModel Logical GLSL450",
+      "OpEntryPoint Vertex %2 \"main\"",
+      "%3 = OpString \"minimal.vert\"",
+      "OpNoLine",
+      "OpLine %3 10 10",
+      "%4 = OpTypeVoid",
+      "OpLine %3 100 100",
+      "%5 = OpTypeFunction %4",
+      "%2 = OpFunction %4 None %5",
+      "OpLine %3 1 1",
+      "OpNoLine",
+      "OpLine %3 2 2",
+      "OpLine %3 3 3",
+      "%6 = OpLabel",
+      "OpLine %3 4 4",
+      "OpNoLine",
+      "OpReturn",
+      "OpLine %3 4 4",
+      "OpNoLine",
+      "OpFunctionEnd",
+  };
+  RunAndCheck(Join(text), SkipDbgInstJoin(text));
+
+  // Let's add more debug instruction before the "OpString" instruction.
+  const std::vector<std::string> more_text = {
+      "OpSourceContinued \"I'm a happy shader! Yay! ;)\"",
+      "OpSourceContinued \"wahahaha\"",
+      "OpSource ESSL 310",
+      "OpSource ESSL 310",
+      "OpSourceContinued \"wahahaha\"",
+      "OpSourceContinued \"wahahaha\"",
+      "OpSourceExtension \"save-the-world-extension\"",
+      "OpName %2 \"main\"",
+      "OpModuleProcessed \"42\"",
+      "OpModuleProcessed \"43\"",
+      "OpModuleProcessed \"44\"",
+  };
+  text.insert(text.begin() + 4, more_text.cbegin(), more_text.cend());
+  RunAndCheck(Join(text), SkipDbgInstJoin(text));
+}
+
+using StripDebugInfoTest = PassTest<::testing::TestWithParam<std::string>>;
+
+TEST_P(StripDebugInfoTest, Kind) {
```

Comment:
This test doesn't access GetParam().  So the parameterized test is essentially a no-op.  :-)